### ISSUE TITLE
fix:  melos can't solve dependencies correctly for dart commands.

### ISSUE THREAD

**Issue by erickzanardo** [invertase/melos#357]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Version

2.6.0

### Description

When using melos on a mono repo that have plain dart packages. It seems that it can't correctly solve the dependencies when a dart package depends on another dart package from the repository.

Imagine the following scenario:
```
my_project
 |- packages
    |- pkg1
    |- pkg2
```

Both `pkg1` and `pkg2` are dart packages, and `pkg2` depends on `pkg1`. When I try to run `dart test` on `pkg2`, I get the following error:

```
Because pkg2 depends on pkg1 any which doesn't exist (could not find package marble at https://pub.dartlang.org), version solving failed.
```

A workaround is using `flutter test` instead, but this isn't optimal since flutter is not needed for a dart package and collicting coverage for dart package through Flutter cli can be problematic (at least I couldn't figure it out 😅 )

### Steps to reproduce

Just create the described scenario on the issue description and try running `dart test` on the `pkg2`.

Or this error can be simulated on this repository: https://github.com/erickzanardo/marble 
Just try running `dart test` on the marble_frog package.

### Expected behavior

dart test should work fine on projects with dependencies managed by melos.

**Comment by blaugold** [comment]
The problem is dart `dart test` is incompatible with the default mechanism used by Melos to override local dependencies.

There is a new mechanism does not have this issue, and you can read more about enabling it [here](https://melos.invertase.dev/configuration/overview#commandbootstrapusepubspecoverrides).

**Comment by blaugold** [comment]
Closing this since it's not an issue with Melos.
